src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -612,6 +612,42 @@ int D4NFilterObject::set_head_obj_dir_entry(const DoutPrefixProvider* dpp, optio
     } else {
       ldpp_dout(dpp, 10) << "D4NFilterObject::" << __func__ << "(): BlockDirectory get method failed for head object with ret: " << ret << dendl;
     }
+  } else if (!this->get_bucket()->versioned()) {
+    rgw::d4n::CacheObj null_object = rgw::d4n::CacheObj{
+      .objName = "_:null_" + this->get_name(),
+      .bucketName = this->get_bucket()->get_name(),
+      .dirty = dirty,
+      .hostsList = { dpp->get_cct()->_conf->rgw_d4n_l1_datacache_address },
+    };
+
+    rgw::d4n::CacheBlock null_block = rgw::d4n::CacheBlock{
+      .cacheObj = null_object,
+      .blockID = 0,
+      .version = this->get_object_version(),
+      .prevVersion = block.prevVersion,
```

Comment:
is there going to be a prevVersion here, or are you just doing it for the sake of completeness?

Reply:
I mean is there a need to do a get and set here for this block? can we not safely assume that null_block.prevVersion = {null_block.version, null_block.deleteMarker}; always?

Reply:
And you haven't set the deleteMarker here, is that intentional and if so why?

Reply:
It was just for the sake of completion but I agree with your point, so I will remove it. In this case, there is also no possibility for a delete marker to occur, correct?

Reply:
Correct.